### PR TITLE
crimson/osd: fix ENOENT on accessing RadosGW user's index of buckets

### PR DESCRIPTION
**Pull request by rzarzynski**
https://gist.github.com/rzarzynski/267661b719c414126a713826175f445c
    
Fixes: https://tracker.ceph.com/issues/59429
Signed-off-by: Radoslaw Zarzynski <rzarzyns@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/osd/osd_operations/client_request.cc [hunk 1]
```diff
@@ -231,11 +231,13 @@ ClientRequest::process_op(instance_handle_t &ihref, Ref<PG> &pg)
           return pg->with_locked_obc(
             m->get_hobj(), op_info,
             [this, pg, &ihref](auto obc) mutable {
-              return ihref.enter_stage<interruptor>(pp(*pg).process, *this
-            ).then_interruptible([this, pg, obc, &ihref]() mutable {
-              return do_process(ihref, pg, obc);
+              logger().debug("{}: got obc {}", *this, obc->obs.exists);
```
Comment (Matan-B) [inline src/crimson/osd/osd_operations/client_request.cc:234]:
```suggestion
              logger().debug("{}: got obc {}", *this, obc->get_oid());
```

**Review by Matan-B — COMMENTED** [review]
If the results are same as in classic, should we add the `s3test.yaml` to the suite?

**Comment by Matan-B** [comment]
If the results of the tests are the same as in classic, should we add `s3tests.yaml` to the crimson suite?

**Comment by athanatos** [comment]
@Matan-B Yeah, once it passes we can go ahead and add it.

**Review by athanatos — APPROVED** [review]

File: src/crimson/osd/osd_operations/client_request.cc (hunk 1 above)
Comment (Matan-B) [reply src/crimson/osd/osd_operations/client_request.cc:234]: I see, `obc->obs` should print both `oid` and `exists` flag. Might be worth using it in order to have better context of the printed boolean in the logs.

Comment (rzarzynski) [reply src/crimson/osd/osd_operations/client_request.cc]: Oops, I guess I now recall ;-) why printed just `exists`:

```
../src/crimson/osd/osd_operations/client_request.cc:216:67:   required from here
../src/fmt/include/fmt/core.h:1757:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1757 |       formattable,
```

Sent a `fmt::formatter<...>` for `ObjectState` as a separated commit as it formally touches the classical thing.

**Comment by Matan-B** [comment]
```
Merge remote-tracking branch 'wip-crimson-ertr-plain-rets' into wip-matanb-crimson-testing-28.6
Merge remote-tracking branch wip-bug-59429' into wip-matanb-crimson-testing-28.6
Merge remote-tracking branch 'wip-matanb-crimson-watch-notify-logs' into wip-matanb-crimson-testing-28.6
Merge remote-tracking branch 'wip-matanb-crimson-no-ignore' into wip-matanb-crimson-testing-28.6
latest main commit 0082f63e8c
```
https://shaman.ceph.com/builds/ceph/wip-matanb-crimson-testing-28.6

Build is ready, awaiting infra failures to resolve.

**Comment by Matan-B** [comment]
Merging based on:
https://pulpito.ceph.com/matan-2023-07-02_07:05:40-crimson-rados-wip-matanb-crimson-testing-28.6-distro-crimson-smithi/

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcBHAY6fGO1801-wip-yuri10-testing-2023-07-10-1345

**Pull request merged**